src/btree/bt_discard.c

```diff
@@ -311,6 +311,32 @@ __wt_ref_addr_free(WT_SESSION_IMPL *session, WT_REF *ref)
         __ref_addr_safe_free(session, ref_addr);
 }
 
+/*
+ * __wt_addr_address_safe_free --
```

Comment:
Why aren't we reusing the [function](https://github.com/wiredtiger/wiredtiger/commit/248864f3ecbe26aaa00ce6696f46d9cff8f481be#diff-362034196a03a3369bc7db1a7200a246e1a7d4b57674981bdac4c438fc63c687R250) that was added in WT-11062?

Reply:
The `__ref_addr_safe_free` function takes care of safe freeing the entire `addr` structure. But in the code flow of `__compact_page_replace_addr`, we are just freeing the `addr->addr`. So we need a new function to handle it.

Reply:
Can we update the existing function to take the following arguments?
```
void *p, size_t len
```
This way, we would update the existing function with:
```
((WT_ADDR *)ref_addr)->addr, ((WT_ADDR *)ref_addr)->size)
```
And then you can adapt it to your needs with:
```
addr->addr, addr->size)
```

Reply:
Thanks; I reorganized the code as per the suggestion. This way the duplicate code is removed.